vm/src/memory/call_stack.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "definitions/function_definition.h"
+#include "locals.h"
+#include "operand_stack.h"
+
+/// Thrown when the memory allocated for a thread call stack
+/// is exceeded
+class StackOverflowError : public MemoryError {
+public:
+    explicit StackOverflowError(std::string message) : MemoryError{message} {}
```

Comment:
```suggestion
/**
 * Thrown when the memory allocated for a thread call stack is exceeded.
 */
class StackOverflowError : public MemoryError {
public:
    explicit StackOverflowError(std::string message) : MemoryError{std::move(message)} {}
```
Don't mix Rust documentation style with Doxygen documentation style.